### PR TITLE
Performing Data Wrangling and EDA (Exploratory Data Analysis)

### PR DESCRIPTION
**Pull request by YolaKamalita**
Resolving #3

Collaborators
@YolaKamalita 
@M-MWW 
@czh000101 

What's remaining?
- Adding description to the codes, graphs, and tables.

What's done?

- Remove notebooks dir, and replace it with quarto dir.
- Move data_processing.Rmd to quarto dir.
- Add scatterplot.R (EDA) to quarto dir.
- Add Group_10_Analysis.qmd to combine data_processing.Rmd and scatterplot.R (EDA), and also add boxplot.

### CODE REVIEW

File: quarto/Group_10_Analysis.qmd [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+-   budget: Budget for the films production (in \$1000000s)
+-   votes: Number of positive votes received by viewers
+-   genre: Genre of the film
+-   rating: IMDB rating from 0-10
+
+# Data Wrangling
+
+## Missing value
+
+```{r}
```
Comment (YolaKamalita) [inline quarto/Group_10_Analysis.qmd:72]: From @M-MWW: Calculate the number of NA, if the proportion is not large, we can remove it, if the number is large, maybe we need to fill with the mean value.

File: quarto/Group_10_Analysis.qmd [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        col = "skyblue",              
+        names.arg = colnames(df), 
+        ylim = c(0, max(na_sum) + 200),
+        las=2,
+        adj=1) 
+
+text(x = 1:length(na_sum)+0.2, y = na_sum + 2, labels = na_sum, pos = 3, cex = 0.8, col = "red")
+```
+
+```{r}
```
Comment (YolaKamalita) [inline quarto/Group_10_Analysis.qmd:104]: From @M-MWW: for each type of movie, select them individually, and then calculate their average rating? then we can know what kind of movies have higher rating.

File: quarto/Group_10_Analysis.qmd (hunk 1 above)
Comment (M-MWW) [reply quarto/Group_10_Analysis.qmd:72]: yes,calculate the number of NA and visualize it

File: quarto/Group_10_Analysis.qmd (hunk 2 above)
Comment (M-MWW) [reply quarto/Group_10_Analysis.qmd:104]: See how many unique values in the categorical variables

**Review by M-MWW — APPROVED** [review]

File: quarto/Group_10_Analysis.qmd [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  geom_boxplot(fill = "steelblue")
+
+p4 <- ggplot(data = df, mapping = aes(x = rating_new, y = votes)) +
+  geom_boxplot(fill = "steelblue")
+
+p5 <- ggplot(df, aes(x= genre,  y = ..prop.., group=rating_new, fill=rating_new)) + 
+  geom_bar(position="dodge", stat="count") +
+  labs(y = "Proportion")
+
+grid.arrange(p1, p2, p3, p4, ncol=2)#This part of the code has reported an error, but I can not solve it, I do not know what the problem is
```
Comment (M-MWW) [inline quarto/Group_10_Analysis.qmd]: This part of the code has reported an error, but I can not solve it, I do not know what the problem is

**Review by M-MWW — APPROVED** [review]

File: quarto/Group_10_Analysis.qmd [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+# Creating summary statistics
+
+# Remove non-numeric columns
+IMDB <- df[, !names(df) %in% c("film_id", "genre", "rating_higher_than_7")]
+
+# Summary statistics with adjusted skim()
+my_skim <- skim_with(base = sfl(n = length), 
+                     numeric = sfl(p0 = NULL, p100 = NULL,hist = NULL))
+knit_print(my_skim(IMDB ))
```
Comment (chelsea7711) [inline quarto/Group_10_Analysis.qmd:139]: Adding genre as a group variable to skim()

File: quarto/Group_10_Analysis.qmd [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+  geom_boxplot(fill = "steelblue") +
+  labs(y = "Budget", x = "Rating > 7")
+
+# Rating > 7 vs. Votes
+p4 <- ggplot(data = df, mapping = aes(x = factor(rating_higher_than_7), y = votes)) +
+  geom_boxplot(fill = "steelblue") +
+  labs(y = "Votes", x = "Rating > 7")
+
+# Rating > 7 vs. Genre
+p6 <- ggplot(df, aes(x= genre,  y = ..prop.., group=rating_higher_than_7, fill=rating_higher_than_7)) +
```
Comment (chelsea7711) [inline quarto/Group_10_Analysis.qmd]: 1. Adjusting the codes (Incomplete pdf output)
2. Adding the legend for barplot

File: quarto/Group_10_Analysis.qmd [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+# Creating correlation heatmap
+corrplot(cor(IMDB), method = "color", 
+         type = "lower", addCoef.col = 'grey')
+```
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Creating scatterplot between rating and explanatory variables
```
Comment (chelsea7711) [inline quarto/Group_10_Analysis.qmd:181]: Adding regression line to the scatterplot.

File: quarto/Group_10_Analysis.qmd [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+     labels = unique_values)
+```
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Create new columns: rating_higher_than_7
+df <- df %>% 
+  mutate(rating_higher_than_7 = ifelse(rating <= 7, 0, 1))
```
Comment (Xiaoyu0077) [inline quarto/Group_10_Analysis.qmd:110]: Set a new column named rating_higher_than_7. If rating is higher than 7 ,the value will be 1, or it will be 0.

**Pull request merged**